scripts/proto-update-check.sh

```diff
@@ -6,23 +6,39 @@ set -ex
 # Warn user to update third_party libraries and push again.
 #
 
-red='\e[0;31m'
-green='\e[0;32m'
-off='\e[0m'
-
 regex='github.com/cosmos/cosmos-sdk|github.com/tendermint/tendermint'
-output=$(git diff HEAD:go.mod..origin/main:go.mod | grep -E -c $regex)
-
-if [[ $output -gt 0 ]]; then
-  echo -e "
-  ${red}A version change was detected in one of the following libraries:
-    - github.com/cosmos/cosmos-sdk
-    - github.com/tendermint/tendermint
-
-  Run the command below to update the third_party proto files and push again.
-    $ make proto-update-deps\n${off}" >&2
-  exit 1
-else
-  echo -e "${green}third_party Protobuf files are up to date.\n${off}"
+output=$(git diff ..origin/main -- go.mod | grep -E -c $regex || true)
+dir="$( cd "$( dirname "${BASH_SOURCE:-$0}" )/.."; pwd -P )"
+
+# single brackets because of `set -e` option above.
+if [ "$output" -gt "0" ]; then
+  echo -e "Downloading latest third_party proto files for comparison..."
+
+  # Download third_party proto files int build/ directory for comparison against $dir /third_party
+  bash "$dir"/scripts/proto-update-deps.sh build
+
+  echo -e "\nChecking Protobuf files for differences...\n"
+
+  DIFF=$(diff -rq -x '*.yaml' --exclude=google "$dir"/build/third_party "$dir"/third_party || true)
+  if [ -n "$DIFF" ]; then
+    echo -e "\n\nDiff log:\n\n$DIFF\n\n"
+
+    echo -e "Found differences in Protobuf files.\n
+    This indicates a version change was detected in one of the following libraries:
+      - github.com/cosmos/cosmos-sdk
+      - github.com/tendermint/tendermint
+
+    Review the diff log above and update accordingly. Perform the following steps locally.
+      1. run: make proto-update-deps
+      2. run: make proto-update-check
+      3. run: diff -rq -x '*.yaml' --exclude=google build/third_party third_party
+      4. commit updates and push\n" >&2
```

Comment:
Here also with the `echo`, `printf`, and `cat << EOF` thing.
```suggestion
    printf '\n\nDiff log:\n\n%s\n\n" "$DIFF"

    cat << EOF >&2
Found differences in Protobuf files.
This indicates a version change was detected in one of the following libraries:
  - github.com/cosmos/cosmos-sdk
  - github.com/tendermint/tendermint

Review the diff log above and update accordingly. Perform the following steps locally.
  1. run: make proto-update-deps
  2. make sure this then passes: make proto-update-check
  3. commit updates and push
EOF
```